### PR TITLE
Feat/typing history

### LINKED ISSUE #85: Typing history
It would be nice to use the up/down arrow to recall typing history in a channel

### PR DESCRIPTION
**Pull request by tarkah**
Resolves #85

### CODE REVIEW

File: src/widget/input.rs [hunk 1]
```diff
@@ -91,8 +104,8 @@ where
                         }
                     };
 
-                    // Clear message, we parsed it succesfully
-                    state.input = String::new();
+                    // Clear message and add it to history
+                    state.history.push_front(std::mem::take(&mut state.input));
```
Comment (neilalexander) [inline src/widget/input.rs:108]: Can this grow unbounded?

Comment (tarkah) [reply src/widget/input.rs:108]: Good point, we should probably have some limit

**Review by casperstorm — APPROVED** [review]
LGTM, great feature.

File: src/widget/input.rs (hunk 1 above)
Comment (tarkah) [reply src/widget/input.rs:110]: 64905a2

I've truncated to 100 here. Shouldn't have much memory footprint and is more than enough.

**Pull request merged**